Automattic/jetpack
# AI Chat block: First pass at chatting with site functionality

## Proposed changes:


This updates the AI Chat block to: 
- Chat with site(bot)
- Basic panel setting
- Registers endpoint for asking about the site
- Registers endpoint for sending feedback about the response

The feature relies on Jetpack Search and a site connection. 

The AI Chat block is in experimental mode and is not displayed to any users currently without a constant set. The wpcom endpoints are limited to a12s. 

This PR does not account for wpcom sites. 

## Jetpack product discussion



p1HpG7-oxA-p2

## Does this pull request change what data or activity we track or use?


No

## Testing instructions:






It should work out of the box with
- A site owned by an a11n (or sandboxed connection for any site)
- Jetpack Search purchased (even the free tier)
- `define( 'JETPACK_BLOCKS_VARIATION', 'beta' );`

Try chatting with the site in the editor. 
Try chatting with the site in the front end. 
Try to modify the "ask" button. 
Try to send feedback. 

You can also test some filterable customizations, such as the answer prompt: 
```
add_filter( 'jetpack_ai_chat_answer_prompt', function() {
    return 'Talk like a cowboy';
} );
```
